karton/core/main.py

```diff
@@ -42,30 +154,82 @@ def process(self, task):
-    parser.add_argument("--config-file", help="Alternative configuration path")
-    args = parser.parse_args()
 
-    config = Config(args.config_file)
+    subparsers = parser.add_subparsers(dest="command", help="sub-command help")
 
-    if args.delete:
-        karton_name = args.delete
-        print(
-            "Are you sure you want to remove binds for karton {karton_name}?\n"
-            "Type in the karton name to confirm".format(karton_name=karton_name)
-        )
-        if input().strip() == karton_name:
-            delete_bind(config, karton_name)
-        else:
-            print("abort")
+    subparsers.add_parser("list", help="List active karton binds")
 
-        return
+    delete_parser = subparsers.add_parser("delete", help="Delete a unused karton bind")
+    delete_parser.add_argument(
+        "-n", "--name", required=True, help="Karton bind identity to remove"
+    )
 
-    if args.list:
-        print_bind_list(config)
-        return
+    configure_parser = subparsers.add_parser(
+        "configure", help="Create a new configuration file"
+    )
+    configure_parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="Overwrite the existing configuration file",
+    )
+
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    if args.command == "configure":
+        config_filename = args.config_file or "./karton.ini"
+
+        log.debug("Creating a new configuration file in %s", config_filename)
+
+        if not args.force and os.path.exists(config_filename):
+            log.error(
+                (
+                    "There's already a configuration file under %s, please delete "
+                    "it or specify a different filename using the -c argument"
+                ),
+                config_filename,
+            )
+            return
+
+        return configuration_wizard(config_filename)
+    else:
+        try:
+            config = Config(args.config_file)
+        except RuntimeError as e:
+            log.error("Error while initializing the karton config: %s", e)
+            log.error(
+                (
+                    "Please correct the configuration file or run `karton configure` "
+                    "to initialize it"
+                )
+            )
+            return
+
+        if args.command == "list":
+            return print_bind_list(config)
+        elif args.command == "delete":
+            karton_name = args.name
+            print(
+                f"Are you sure you want to remove binds for karton {karton_name}?\n"
+                "Type in the karton name to confirm deletion."
+            )
+            if input().strip() == karton_name:
+                return delete_bind(config, karton_name)
+            else:
+                log.info("abort")
+                return
+        else:
```

Comment:
you could also give a list of choices to the argparse command, but this is nice too